src/structures/gdt.rs

```diff
@@ -10,27 +10,44 @@ use core::fmt;
 #[cfg(doc)]
 use crate::registers::segmentation::{Segment, CS, SS};
 
+#[cfg(feature = "instructions")]
+use core::sync::atomic::{AtomicU64, Ordering};
+#[cfg(not(feature = "instructions"))]
+type AtomicU64 = u64;
```

Comment:
I'm not sure if that's a pattern we should use. I'd be fine alias importing this:
```suggestion
#[cfg(not(feature = "instructions"))]
use core::primitive::u64 as EntryValue;
#[cfg(feature = "instructions")]
use core::sync::atomic::{AtomicU64 as EntryValue, Ordering};
```

Reply:
That's very fair. I think it could be even furthur simplified to:
```rust
#[cfg(not(feature = "instructions"))]
use u64 as EntryValue;
#[cfg(feature = "instructions")]
use core::sync::atomic::{AtomicU64 as EntryValue, Ordering};
```